metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowPostProcessor.java

```diff
@@ -105,8 +115,21 @@ private void indexPostProcess(AbstractExecutablePlugin<?> indexPlugin, String da
     } else if (indexPlugin instanceof IndexToPublishPlugin) {
       targetIndexingDatabase = ((IndexToPublishPlugin) indexPlugin).getTargetIndexingDatabase();
       targetIndexingEnvironment = ((IndexToPublishPlugin) indexPlugin).getTargetIndexingEnvironment();
-      //Reset depublish status
-      depublishRecordIdDao.markRecordIdsWithDepublicationStatus(datasetId, null,
+
+      // get all tasks from dataset id and topology name
+      List<SubTaskInfo> taskReport = dpsClient.getDetailedTaskReport(indexPlugin.getTopologyName(),
```

Comment:
I don't think we should be doing it this way.
We are requesting here all the datasetIds with one call from dps, right?
What if the ids are in the millions or hundreds of thousands?
This not only can have a huge impact in performance, even timeout on the request but also memory consumption.

A better way would be to send the ids that we are interested in and dps should tell us if those were affected on it's response.

Reply:
Discussed that it will be a new implementation MET-4321, we will use this one as first approach to watch how it performs.